src/hotspot/os/linux/os_linux.cpp

```diff
@@ -1601,14 +1601,13 @@ void * os::dll_load(const char *filename, char *ebuf, int ebuflen) {
   }
 
   Elf32_Ehdr elf_head;
-  int diag_msg_max_length=ebuflen-strlen(ebuf);
-  char* diag_msg_buf=ebuf+strlen(ebuf);
-
-  if (diag_msg_max_length==0) {
+  int diag_msg_max_length = ebuflen - checked_cast<int>(strlen(ebuf));
+  if (diag_msg_max_length <= 0) {
     // No more space in ebuf for additional diagnostics message
     return nullptr;
   }
```

Comment:
```suggestion
  size_t ebuf_len = strlen(ebuf);
  if (ebuflen <= ebuf_len) {
    // No more space in ebuf for additional diagnostics message
    return nullptr;
  }
  size_t diag_msg_max_length = ebuflen - ebuf_len;
```

Reply:
/scratch/cphillim/hg/22os-conversion/src/hotspot/os/linux/os_linux.cpp:1605:15: warning: comparison of integer expressions of different signedness: 'int' and 'size_t' {aka 'long unsigned int'} [-Wsign-compare]
 1605 |   if (ebuflen <= ebuf_len) {
      |       ~~~~~~~~^~~~~~~~~~~

Reply:
Sorry, I thought ebuflen was size_t.